plugins/inputs/docker/client.go

```diff
@@ -27,29 +26,26 @@ type Client interface {
 }
 
 func NewEnvClient() (Client, error) {
-	client, err := docker.NewEnvClient()
+	client, err := docker.NewClientWithOpts(docker.FromEnv)
 	if err != nil {
 		return nil, err
 	}
 	return &SocketClient{client}, nil
 }
 
 func NewClient(host string, tlsConfig *tls.Config) (Client, error) {
-	proto, addr, _, err := docker.ParseHost(host)
-	if err != nil {
-		return nil, err
-	}
-
 	transport := &http.Transport{
 		TLSClientConfig: tlsConfig,
 	}
-	sockets.ConfigureTransport(transport, proto, addr)
 	httpClient := &http.Client{Transport: transport}
 
-	client, err := docker.NewClient(host, version, httpClient, defaultHeaders)
+	client, err := docker.NewClientWithOpts(docker.WithHTTPHeaders(defaultHeaders), docker.WithHTTPClient(httpClient), docker.WithVersion(version), docker.WithHost(host))
 	if err != nil {
 		return nil, err
 	}
+
+	client.NegotiateAPIVersion(context.Background())
```

Comment:
Does this override the initial version above?

Reply:
Actually no, it apparently only updates if the server api version is lower than the client version (specified above) since 1.24 is the lowest reported server api version, `NegotiateAPIVersion` doesn't do anything for us. I'll remove it